compiler/ballerina-lang/src/main/java/io/ballerina/projects/BuildOptions.java

```diff
@@ -71,6 +73,10 @@ public boolean sticky() {
         return this.compilationOptions.sticky();
     }
 
+    public boolean disableSyntaxTree() {
```

Comment:
Let's make this a compilation option. BuildOptions will be suggested in the Ballerina.toml in VSCode AFAIR.